### PR TITLE
Backport rubocop changes

### PR DESCRIPTION
**Pull request by varyonic**
Backport #5152 #5245 #5268 #5138 #5162

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -171,3 +162,33 @@ A PR can only be merged into master by a maintainer if:
...
+  ```sh
+  git checkout master
+  git fetch activeadmin
+  git rebase activeadmin/master
+  # If the release is 2.1.x then this should be: 2-1-stable
+  git checkout -b N-N-stable
+  git push activeadmin N-N-stable:N-N-stable
+  ```
+
+* `bundle exec rake release`
```
[the remark this answers is not in the document]
Comment (varyonic) [reply CONTRIBUTING.md]: This was #5098 in its entirety. I propose we leave it included now.

File: lib/active_admin/views/index_as_grid.rb [hunk 2]
```diff
@@ -42,7 +42,7 @@ def self.index_name
         "grid"
       end
 
-     protected
+      protected
```
[the remark this answers is not in the document]
Comment (varyonic) [reply lib/active_admin/views/index_as_grid.rb]: https://github.com/activeadmin/activeadmin/pull/5162/files includes six files, all are included here as far as I see.

File: spec/support/rails_template_with_data.rb [hunk 3]
```diff
@@ -269,7 +269,7 @@
   end
 RUBY
 
-inject_into_file 'app/admin/tags.rb', <<-RUBY, after: "ActiveAdmin.register Tag do\n"
+inject_into_file 'app/admin/tag.rb', <<-RUBY, after: "ActiveAdmin.register Tag do\n"
```
[the remark this answers is not in the document]
Comment (varyonic) [reply spec/support/rails_template_with_data.rb]: You're right, I'll re-pick this.

**Comment by varyonic** [comment]
I removed the unintended changes to rails_template_with_data. I can remove the extra bullet point from CONTRIBUTING if required, let me know.

File: CONTRIBUTING.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (deivid-rodriguez) [reply CONTRIBUTING.md]: Ok, up to you and @javierjulio if you think it's worth restoring the proper commit so I don't "steal credit"!

File: spec/support/rails_template_with_data.rb [hunk 4]
```diff
@@ -318,8 +318,8 @@
     ActiveAdmin::Comment.create!(
       namespace: :admin,
       author: AdminUser.first,
-      body: "Test comment #{i}",
-      resource: categories.sample,
+      body: "Test comment \#{i}",
```
Comment (deivid-rodriguez) [inline spec/support/rails_template_with_data.rb]: We're also backporting, probably unintentionally, https://github.com/activeadmin/activeadmin/pull/5136 here. I don't have a problem with it but I think backporting the proper commit can make git history a bit less obscure.

**Comment by varyonic** [comment]
Removed unintended #5098 and #5136

**Comment by varyonic** [comment]
@deivid-rodriguez I think enabling CodeClimate on this branch is easiest for now.

**Comment by deivid-rodriguez** [comment]
@varyonic Ok, I'll deal with that and get back here when I'm done!

**Comment by deivid-rodriguez** [comment]
I think it's done now, @varyonic! Can you amend a commit message or push an empty commit to re-trigger the checks?

**Comment by varyonic** [comment]
Amended.

**Comment by deivid-rodriguez** [comment]
Sorry @varyonic, could you do that once again now? I just triggered analysis of the base commit in the 1.4-stable branch (see https://codeclimate.com/github/activeadmin/activeadmin/branches). Maybe the check will be triggered now that there's a base report.

If this doesn't work, let's ignore it for now and merge anyways.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -171,3 +169,18 @@ A PR can only be merged into master by a maintainer if:
 
 Any maintainer is allowed to merge a PR if all of these conditions are
 met.
+
+* `bundle exec rake release`
```
Comment (deivid-rodriguez) [inline CONTRIBUTING.md]: I just detected this unintended addition, we can force push this commit and use it to see if codeclimate retriggers.

**Pull request merged**